shell/platform/android/io/flutter/plugin/common/BinaryMessenger.java

```diff
@@ -139,8 +139,8 @@ default TaskQueue makeBackgroundTaskQueue(TaskQueueOptions options) {
   @UiThread
   default void setMessageHandler(
       @NonNull String channel,
-      @Nullable BinaryMessageHandler handler,
-      @Nullable TaskQueue taskQueue) {
+      @Nullable TaskQueue taskQueue,
+      @Nullable BinaryMessageHandler handler) {
```

Comment:
this is certainly a breaking change :/

Reply:
Right, I wanted your opinion on this. There is a family of kotlin compatibility checks that enforces lambdas to be at the end

Reply:
I could just ignore this lint as it the only way to resolve it is to follow its suggestions

Reply:
this is pretty severe change. I think we would need to disable this specific linter check, and file an issue for modernizing the embedding, so it's more Kotlin friendly. :)